### PR TITLE
Code Improvement: Added TSDoc Linting and Comments

### PR DESCRIPTION
**Pull request by Fritsch-Tech**
#### 🔧 Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update
- [x] Codebase improvement
- [ ] Other (if none of the other choices apply)

#### 🚨 Checklist

- [x] I've read the [guidelines for contributing](https://github.com/gremo/react-directus/blob/main/CONTRIBUTING.md)
- [x] I've added necessary documentation (if appropriate)
- [x] I've ensured that my code additions do not fail linting or unit tests (if applicable)

#### Description

This pull request introduces two significant enhancements to the codebase:

1. **Linting for TSDoc Comments**: I have added TSDoc comment linting to ESLint, ensuring that all TSDoc comments adhere to consistent and proper formatting. This improvement will enhance code readability and maintainability by enforcing a standard for documentation comments.

2. **Added TSDoc Comments**: I have also included TSDoc comments throughout the codebase to provide comprehensive and self-explanatory documentation for various functions, methods, and components. These comments will serve as a valuable resource for developers, promoting better understanding and reducing the learning curve.

Together, these changes will improve code quality and facilitate smoother collaboration among contributors.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,6 +41,7 @@
     "eslint": "8.46.0",
     "eslint-config-prettier": "8.9.0",
     "eslint-plugin-prettier": "5.0.0",
+    "eslint-plugin-tsdoc": "^0.2.17",
```
Comment (gremo) [inline package.json]: Pin this otherwise renovate bot will create a PR

**Comment by gremo** [comment]
Nice, that's what I was asking for before the doc automatic generation 👍 

Does changing @default to `@defaultValue` break something? Or IDE completion/help will work with TSDoc without problems?

**Comment by Fritsch-Tech** [comment]
No, it was actualy just a typo.

File: src/hooks/useDirectusAuth.tsx
Comment (gremo) [inline src/hooks/useDirectusAuth.tsx:1]: This not belongs to this PR right? It's a fix for a previous mistake?

Comment (Fritsch-Tech) [reply src/hooks/useDirectusAuth.tsx:1]: I added it here to make shure all the TSDoc comments i wrote are in line with the standard.

Comment (gremo) [reply src/hooks/useDirectusAuth.tsx:1]: Sorry for the mistake, I mean the change from `..` to `@` of the imports!

Comment (Fritsch-Tech) [reply src/hooks/useDirectusAuth.tsx:1]: You are right, when switchig branches I brought the changes with me.
I can remove it again if you like.

Comment (gremo) [reply src/hooks/useDirectusAuth.tsx:1]: Better to not mix fixes with this PR

**Comment by gremo** [comment]
Thanks!

**Pull request merged**